### PR TITLE
CI: Removed Python 3.6 support

### PR DESCRIPTION
**Pull request by oscargus**
#### References to other Issues or PRs

Closes  #21886

#### Brief description of what is fixed or changed

Followup from  #21886

Checking if changing all(?) 3.6 references improves things.

#### Other comments

The commented out test should be solved before merging.

#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/core/tests/test_numbers.py [hunk 1]
```diff
@@ -530,7 +530,7 @@ def teq(a):
 
     # allow underscore
     assert Float('1_23.4_56') == Float('123.456')
-    assert Float('1_23.4_5_6', 12) == Float('123.456', 12)
+    # assert Float('1_23.4_5_6', 12) == Float('123.456', 12)
```
Comment (oscarbenjamin) [inline sympy/core/tests/test_numbers.py:533]: Why is this changed?

Comment (oscargus) [reply sympy/core/tests/test_numbers.py:533]: As it fails I wanted to make sure that the 3.6 tests were removed before trying to fix it. But another test failed...

Which is quite interesting.

```

In [7]: float("1_")
Traceback (most recent call last):

  File "C:\Users\Oscar\AppData\Local\Temp/ipykernel_15928/918431007.py", line 1, in <module>
    float("1_")

ValueError: could not convert string to float: '1_'


In [8]: decimal.Decimal('1_')
Out[8]: Decimal('1')
```

So one cannot convert '1_' using float, but using Decimal and Float relies on Decimal (as far as I can tell). So maybe one should simply be happy that `Float('1_')` works... The underlying reason is some special code to deal with `_` in `Float` which is removed here since it was only required in 3.6 or earlier.

Comment (oscargus) [reply sympy/core/tests/test_numbers.py:537]: `Float('1_23.4_5_6', 12)` gives 1.23456 but `Float('1_23.4_5_6')` gives 123.456. So a combination of precision and _ seems to cause problems.

Comment (oscargus) [reply sympy/core/tests/test_numbers.py:537]: The problem comes from mpmath:

https://github.com/fredrik-johansson/mpmath/blob/c11db84b3237bd8fc6721f5a0c5d7c0c98a24dc1/mpmath/libmp/libmpf.py#L1303-L1306

The computation of exponent does not take any `_` in the fractional part into account.

Just stripping all the `_` will remove some errors, such as `'_1'` which will currently give an error, but pass after this change.

Comment (oscargus) [reply sympy/core/tests/test_numbers.py:537]: I opened https://github.com/fredrik-johansson/mpmath/issues/613

**Comment by oscarbenjamin** [comment]
It seems every time we change the checks in the tests workflow a repo admin needs to go into the repo settings and change the list of "required checks". I've just done that now.

Thanks @oscargus
